Sources/AppcuesKit/Presentation/Actions/Appcues/AppcuesTrackAction.swift

```diff
@@ -10,27 +10,67 @@ import Foundation
 
 @available(iOS 13.0, *)
 internal class AppcuesTrackAction: AppcuesExperienceAction {
-    struct Config: Decodable {
+    struct Config {
         let eventName: String
+        let attributes: [String: Any]?
     }
 
     static let type = "@appcues/track"
 
     private weak var appcues: Appcues?
 
     let eventName: String
-
+    let attributes: [String: Any]?
+    
     required init?(configuration: AppcuesExperiencePluginConfiguration) {
         self.appcues = configuration.appcues
 
         guard let config = configuration.decode(Config.self) else { return nil }
         self.eventName = config.eventName
+        self.attributes = config.attributes
     }
 
     func execute(completion: ActionRegistry.Completion) {
         guard let appcues = appcues else { return completion() }
 
-        appcues.track(name: eventName)
+        appcues.track(name: eventName, properties: attributes)
         completion()
     }
 }
+
+@available(iOS 13.0, *)
+extension AppcuesTrackAction.Config: Decodable {
+    // Custom decoding for this one - we want to just gather up all the key/value
+    // pairs from the trait config as raw user property values. We trim the set to
+    // only those of supported data types and then store in the resulting properties
+    // dictionary
+    init(from decoder: Decoder) throws {
+               
+        let container = try decoder.container(keyedBy: DynamicCodingKeys.self)
+
+        self.eventName = try container.decode(String.self, forKey: DynamicCodingKeys(stringValue: "eventName")!)
+        
+        var dict: [String: Any] = [:]
+
+        container.allKeys.forEach { key in
+            if key.stringValue == "eventName" {
+                // If it's the eventName, skip it
+                return
```

Comment:
"eventName" is already decoded on L51, I think there is some way to just pull out "attributes" as its own container, then loop through its items?

Reply:
actually this is probably not right. I will take another look!

Reply:
If you use the CodingKey approach I suggested above:

```swift
var attributesContainer = container.nestedContainer(keyedBy: DynamicCodingKeys.self, forKey: .attributes)
```

Reply:
working through that!